packages/state-transition/src/block/processDeposit.ts

```diff
@@ -62,37 +56,42 @@ export function applyDeposit(
   deposit: DepositData | DepositRequest
 ): void {
   const {config, validators, epochCtx} = state;
-  const {pubkey, withdrawalCredentials, amount} = deposit;
+  const {pubkey, withdrawalCredentials, amount, signature} = deposit;
 
   const cachedIndex = epochCtx.getValidatorIndex(pubkey);
-  if (cachedIndex === null || !Number.isSafeInteger(cachedIndex) || cachedIndex >= validators.length) {
-    if (isValidDepositSignature(config, pubkey, withdrawalCredentials, amount, deposit.signature)) {
-      addValidatorToRegistry(fork, state, pubkey, withdrawalCredentials, amount);
-    }
-  } else {
-    if (fork < ForkSeq.electra) {
+  const isNewValidator = cachedIndex === null || !Number.isSafeInteger(cachedIndex) || cachedIndex >= validators.length;
```

Comment:
`getValidatorIndex` only returns null or a value, I think just base on that is enough
at least `cachedIndex >= validators.length` should be removed, or if it happens we throw error

Reply:
> if it happens we throw error

Don't think this will happen after we remove unfinalized pubkey cache. 

I will just remove the check